cmd/prometheus/main.go

```diff
@@ -97,6 +97,50 @@ func init() {
 	}
 }
 
+type flagConfig struct {
+	configFile string
+
+	localStoragePath    string
+	notifier            notifier.Options
+	notifierTimeout     model.Duration
+	forGracePeriod      model.Duration
+	outageTolerance     model.Duration
+	resendDelay         model.Duration
+	web                 web.Options
+	tsdb                tsdbOptions
+	lookbackDelta       model.Duration
+	webTimeout          model.Duration
+	queryTimeout        model.Duration
+	queryConcurrency    int
+	queryMaxSamples     int
+	RemoteFlushDeadline model.Duration
+
+	featureList []string
+	// These options are extracted from featureList
+	// for ease of use.
+	enablePromQLAtModifier bool
+
+	prometheusURL   string
+	corsRegexString string
+
+	promlogConfig promlog.Config
+}
+
+// setFeatureListOptions sets the corresponding options from the featureList.
+func (c *flagConfig) setFeatureListOptions() error {
+	for _, f := range c.featureList {
+		switch f {
+		case "promql-at-modifier":
+			c.enablePromQLAtModifier = true
+		case "":
+			continue
+		default:
+			return errors.Errorf("invalid feature %q", f)
```

Comment:
This shouldn't be an error, a log line at most. Being able to referring to a feature that is only in a future/past version makes rollouts easier (plus it's what was said at the dev summit).

Reply:
Yes, a warning would be nicer.

Reply:
It will log a warning now.